typescript/web/src/components/dataset-class-list/class-list.tsx

```diff
@@ -105,56 +117,35 @@ export const ClassesList = ({
         labelClassId={deleteClassId}
         onClose={() => setDeleteClassId(null)}
       />
-
-      <Box
-        d="flex"
-        flexDirection="column"
-        bg={mode("white", "gray.800")}
-        m={{ base: "2", md: "8" }}
-        borderRadius="lg"
-        maxWidth="5xl"
-        flexGrow={1}
-      >
-        <>
-          <Text
-            margin="2"
-            fontWeight="bold"
-          >{`${labelClassWithShortcut.length} Classes`}</Text>
-          <Divider />
-          {!loading && (
-            <DragDropContext onDragEnd={onDragEnd}>
-              <Droppable droppableId="droppable">
-                {(provided) => (
-                  <Box
-                    {...provided.droppableProps}
-                    ref={provided.innerRef}
-                    // style={getListStyle(snapshot.isDraggingOver)}
-                  >
-                    {labelClassWithShortcut.map(
-                      ({ id, name, color, shortcut, index }) => (
-                        <ClassItem
-                          key={id}
-                          id={id}
-                          index={index}
-                          name={name}
-                          color={color}
-                          shortcut={shortcut}
-                          edit={editClassId === id}
-                          datasetSlug={datasetSlug}
-                          workspaceSlug={workspaceSlug}
-                          onClickEdit={setEditClassId}
-                          onClickDelete={setDeleteClassId}
-                        />
-                      )
-                    )}
-                    {provided.placeholder}
-                  </Box>
-                )}
-              </Droppable>
-            </DragDropContext>
-          )}
-        </>
-      </Box>
+      <UpsertClassModal
+        isOpen={editClassId != null}
+        classId={editClassId}
+        onClose={() => setEditClassId(null)}
+        datasetId={datasetId}
+        datasetSlug={datasetSlug}
+      />
+      {!loading && (
+        <Box display="flex" flexDirection="column" w="full" p={8}>
+          <Heading
+            mb={5}
+          >{`Classes (${labelClassWithShortcut.length})`}</Heading>
+          <ClassTableActions
+            searchText={searchText}
+            setSearchText={setSearchText}
+            isCreatingClassLabel={isCreatingClassLabel}
+            setIsCreatingClassLabel={setIsCreatingClassLabel}
+            datasetId={datasetId}
+            datasetSlug={datasetSlug}
+          />
+          <ClassTableContent
+            classes={labelClassWithShortcut}
+            onDragEnd={onDragEnd}
+            onClickDelete={setDeleteClassId}
+            onClickEdit={setEditClassId}
+            searchText={searchText}
+          />
+        </Box>
+      )}
```

Comment:
I'm not sure that I understand why you want to split this component into smaller parts ? If I understood your extension correctly, it will increase the complexity as the number of components will increase as well